mod/solr_api/start.php

```diff
@@ -0,0 +1,302 @@
+        'get.entity_list',
+        'get_entity_list',
+        [
+            'type' => [
+                'type' => 'string',
+                'required' => true,
+                'description' => 'the type of entity in string format',
+            ],
+            'subtype' => [
+                'type' => 'string',
+                'required' => false,
+                'description' => 'the subtype of entity in string format, not required',
+            ],
+			'offset' => [
+                'type' => 'int',
+                'required' => true,
+                'description' => 'the subtype of entity in string format, not required',
+            ],
+
+        ],
+        'retrieves all entities filtered by type [and subtype]',
+        'GET',
+        true,
+        false
+	);
+
+
+	elgg_ws_expose_function(
+        'get.user_list',
+        'get_user_list',
+        [
+        	'offset' => [
+        		'type' => 'int',
+        		'required' => true,
+        		'description' => 'paging mechanism'
+        	]
+        ],
+        'retrieves a user list',
+        'GET',
+        true,
+        false
+	);
+
+	elgg_ws_expose_function(
+        'get.group_list',
+        'get_group_list',
+        [
+        	'offset' => [
+        		'type' => 'int',
+        		'required' => true,
+        		'description' => 'api loads 20 groups at a time'
+        	]
+        ],
+        'retrieves a group list',
+        'GET',
+        true,
+        false
+	);
+
+
+	elgg_ws_expose_function(
+		'get.list_of_deleted_records',
+		'get_list_of_deleted_records',
+		null,
+		'retrieves a list of deleted content',
+		'GET',
+		true,
+		false
+	);
+}
+
+
+function delete_updated_index_list($guids) {
+
+	$ids = array();
+	foreach ($guids as $guid) {
+		// guid must be an integer
+		if (is_numeric($guid)) 
+			$ids[] = $guid;
+		else
+			return "there is an issue deleting a record in the database, please see function delete_updated_index_list()";
+	}
+
+	if (sizeof($ids) > 0) {
+		$ids = implode(",", $ids);
+		$query = "DELETE FROM deleted_object_tracker WHERE id IN ({$ids})";
+		$result = delete_data($query);
+	}
+
+	$arr[] = array('message' => 'deleted the following guid');
+	$arr[] = $ids;
+
+	return $arr;
+}
+
+// this is not an api call, this is a hook that will initiate once user attempts deletion
+function intercept_object_deletion($event, $type, $object) {
+	$unixtime = time();
+
+	$query = "INSERT INTO deleted_object_tracker (id, time_deleted) VALUES ({$object->guid}, {$unixtime})";
+
+	// just in case for some reason, the user is able to delete the same thing twice
+	try {
+		$insert_record = insert_data($query);
+	} catch (Exception $e) {
+		error_log("error occurred: {$e}");
+	}
+
+	$query = "SELECT * FROM deleted_object_tracker";
+	$deleted_records = get_data($query);
+
+	foreach ($deleted_records as $deleted_record) {
+		error_log("the id: " . $deleted_record->id);
+	}
+
+
+	return false;
+}
+
+
+// api calls with sample set of limit 15
```

Comment:
Where's the limit 15? most seem to have a limit of 10.

Reply:
oh, oops. i forgot to update the comments, i had planned to default to a limit of 15 but realized the results could be too much so i changed it to 10